OrbitLinuxTracing/PerfEventReaders.cpp

```diff
@@ -87,4 +87,12 @@ std::unique_ptr<CallchainSamplePerfEvent> ConsumeCallchainSamplePerfEvent(
   return event;
 }
 
+std::unique_ptr<TracepointEventPidTidTimeCpu> ConsumeTracepointEventPidTipCpuTime(
+    PerfEventRingBuffer* ring_buffer, const perf_event_header& header) {
+  auto event = std::make_unique<TracepointEventPidTidTimeCpu>();
+  ring_buffer->ReadRawAtOffset(&event->ring_buffer_record, sizeof(perf_event_header),
+                               sizeof(perf_event_sample_id_tid_time_streamid_cpu));
+  ring_buffer->SkipRecord(header);
+  return event;
+}
```

Comment:
If you read the entire record, you can use `PerfEventRingBuffer::ConsumeRecord` directly, without creating a custom method here. See usages of that method.

Reply:
Thank you!

Reply:
PerfEventRingBuffer::ConsumeRecord would not read everything that was in the ring buffer? Because I have filled the ring buffer with all the information required for the tracepoints (including the header and format specific data) and I am interested in reading only a small portion of it ( perf_event_sample_id_tid_time_streamid_cpu).

Reply:
Discussed offline, as the record also contains the raw tracepoint data but we don't want to read it or store it here, a custom reader is needed.